ibmcloud/terraform/cluster/main.tf

```diff
@@ -14,10 +14,23 @@ locals {
   bastion_ip = resource.ibm_is_floating_ip.worker.address
 }
 
+resource "ibm_is_ssh_key" "created_ssh_key" {
+  # Create the ssh key only if the public key is set
+  count = var.ssh_pub_key == "" ? 0 : 1
+  name = var.ssh_key_name
+  public_key = var.ssh_pub_key
+}
+
 data "ibm_is_ssh_key" "ssh_key" {
+  # Wait if the key needs creating first
+  depends_on = [ibm_is_ssh_key.created_ssh_key]
   name = var.ssh_key_name
 }
 
+output "ssh_key_name" {
```

Comment:
This is an optional change, but it's recommended to put Terraform `output`s into a separate `outputs.tf` file, [per the source here](https://learn.hashicorp.com/tutorials/terraform/outputs#output-vpc-and-load-balancer-information).